src/components/UploadButton/UploadButton.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import './UploadButton.css';
+
+interface PropTypes {
+  // Button contents
+  label?: string;
+  // This type comes from a built-in React method.
+  // Essentially, its typed to the "setState" function.
+  setFile: React.Dispatch<React.SetStateAction<string | null>>;
+};
+
+// Primary UI component for user interaction
+const UploadButton = ({
+  label="Upload a File...",
+  setFile,
+  ...props
```

Comment:
What other props here? You have exhausted all the declared props for `PropTypes`. 

If you want to allow the caller to pass other props, you should list them. Or you can do `interface PropTypes extends React.HTMLAttributes<HTMLInputElement>`. And if you want to limit which props can be set, you can use `Pick` or `Omit`.

For example:

```
interface PropTypes extends Pick<React.HTMLAttributes<HTMLInputElement>, 'readonly' | 'disabled'> {
...
}
```

Reply:
I've extended the Omit / HTMLProps / HTMLInputElement like you mentioned here... You can see those changes in https://github.com/AsyncRacing/AsyncRacing/pull/5/commits/3acc4ea37502b5dafe20e91a2ba7190b41d1839f . Just wanted to make sure this is correct before resolving this conversation. Let me know if it looks right!

Reply:
Specifically, I did *not* remove the spread `{,,,props}` from the component. Is this best practice?

Reply:
lgtm

Reply:
After looking again, I think if it was my project I would not allow passing more props here. The reason is that the `input` is not the root of what you render. Someone might pass some properties (like style or class) and expect them to be applied directly but because you are wrapping in a div and adding a label it will definitely not behave like a standard input element. 

I would lean in favor of keeping things simple and not accepting more props than are strictly needed at the moment. 

This is what I would do but definitely not a blocking comment though.

Reply:
With https://github.com/AsyncRacing/AsyncRacing/pull/5/commits/1fb6c86b0807bed31fff6914e135287f890d2ac7 :
The input is now the root of what the component renders. I took out the label element so it can be added elsewhere, and kept all the proptypes and spread. It would most likely still be easier to remove the spread like you're saying though.

Reply:
Resolved with https://github.com/AsyncRacing/AsyncRacing/pull/5/commits/6d5fc55ee80235e4a3ecac835b8af24292a4741a